pandas-stubs/core/series.pyi

```diff
@@ -3821,23 +3825,23 @@ class Series(IndexOpsMixin[S1], NDFrame):
     ) -> Series[S1]: ...
     @overload
     def cumprod(
-        self: Series[_str],
+        self: Series[Never],
         axis: AxisIndex = ...,
         skipna: _bool = ...,
         *args: Any,
         **kwargs: Any,
-    ) -> Never: ...
+    ) -> Series: ...
     @overload
     def cumprod(
-        self: Series[Timestamp],
+        self: Series[bool],
         axis: AxisIndex = ...,
         skipna: _bool = ...,
         *args: Any,
         **kwargs: Any,
-    ) -> Never: ...
+    ) -> Series[int]: ...
     @overload
     def cumprod(
-        self,
+        self: SupportsGetItem[Scalar, _SupportsMul[S1]],
```

Comment:
Does that work with classes that have overloads for `__mul__`? My understanding from https://github.com/microsoft/pyright/discussions/6549#discussioncomment-7669091 is that type checkers will then pick the first overload. It might be okay in this case (and I wish we could use this pattern!) - just something to be aware of when using this patterns in more places.

Reply:
Not sure that it would matter, because `S1` are types we define in `pandas` as well as fundamental types (`int`, `float`, etc.), so I think we will be fine.  It's a bit different than the case you wrote up, because of the use of `_SupportsMul[S1]` requiring the type that supports `__mul__()` to be in `S1`